enso/resample/__init__.py

```diff
@@ -42,7 +41,58 @@ def resample(X, y, max_ratio=50):
 
         random.shuffle(idx_sample)
 
-        return X[idx_sample].tolist(), y[idx_sample].tolist()
+        return X.tolist() + X[idx_sample].tolist(), y.tolist() + y[idx_sample].tolist()
+
+
+@Registry.register_resampler(ModeKeys.CLASSIFY)
+class CorruptiveResampler(Resampler):
+
+    @staticmethod
+    def scramble(y, prob):
+        classes = list(set(y))
+        y_ = []
+        baseline_probas = np.ones(len(classes)) * prob / (len(classes) - 1)
+        true_proba = 1 - sum(baseline_probas)
+        for sample in y:
+            bl = np.copy(baseline_probas)
+            bl[classes.index(sample)] += true_proba
+            y_.append(np.random.choice(classes, p=bl))
+        return y_
+
+    @staticmethod
+    def resample(X, y, max_ratio=50):
+        num_samples = len(y)
+        num_gold = int(num_samples * GOLD_FRAC)
+
+        y_gold = y[:num_gold]
+        y_train = np.asarray(list(y_gold) + CorruptiveResampler.scramble(y[num_gold:], CORRUPTION_FRAC))
+        return X, y_train
+
+
+@Registry.register_resampler(ModeKeys.CLASSIFY)
+class CorruptiveOversampler(Resampler):
+
+    @staticmethod
+    def resample(X, y, max_ratio=50):
+        X, y = CorruptiveResampler.resample(X, y)
+        return RandomOverSampler.resample(X, y)
+
+@Registry.register_resampler(ModeKeys.CLASSIFY)
+class CorruptiveOversamplerWithOriginalTargs(Resampler):
```

Comment:
I think we may be able to rip these two classes out? IIRC it kinda breaks the enso abstractions a bit with the modification to X.